include/ck/utility/type_convert.hpp

```diff
@@ -9,15 +9,25 @@
 
 namespace ck {
 
-// Convert X to Y
-template <typename Y, typename X>
+// Convert X to Y, Y is a non-const data type.
+template <typename Y, typename X, std::enable_if_t<!std::is_const_v<Y>, bool> = false>
 __host__ __device__ constexpr Y type_convert(X x)
 {
     static_assert(!std::is_reference_v<Y> && !std::is_reference_v<X>);
 
     return static_cast<Y>(x);
 }
 
+// Convert X to Y, Y is a const data type.
+template <typename Y, typename X, std::enable_if_t<std::is_const_v<Y>, bool> = false>
```

Comment:
What if X is const? Is it handled?

Reply:
In our cases, when we write `type_convert<const DataType>(x)` it works well even if `x` is a const variable since it's passed by value and captured as a non const copy.

But after seeing your comment I started thinking that a user can directly specify the datatype of `X` as const like `type_convert<const DataType, const OtherDataType>(x)` (no reason to do that tbh, but still possible) and it wouldn't work. So I extended it to handle also such cases.